include/session/config/convo_info_volatile.h

```diff
@@ -0,0 +1,219 @@
+
+/// Adds or updates a conversation from the given convo info
+void convo_info_volatile_set_1to1(config_object* conf, const convo_info_volatile_1to1* convo);
+void convo_info_volatile_set_open(config_object* conf, const convo_info_volatile_open* convo);
+void convo_info_volatile_set_legacy_closed(
+        config_object* conf, const convo_info_volatile_legacy_closed* convo);
+
+/// Erases a conversation from the conversation list.  Returns true if the conversation was found
+/// and removed, false if the conversation was not present.  You must not call this during
+/// iteration; see details below.
+bool convo_info_volatile_erase_1to1(config_object* conf, const char* session_id);
+bool convo_info_volatile_erase_open(
+        config_object* conf, const char* base_url, const char* room, unsigned const char* pubkey);
+bool convo_info_volatile_erase_legacy_closed(config_object* conf, const char* group_id);
+
+/// Returns the number of conversations.
+size_t convo_info_volatile_size(const config_object* conf);
+/// Returns the number of conversations of the specific type.
+size_t convo_info_volatile_size_1to1(const config_object* conf);
+size_t convo_info_volatile_size_open(const config_object* conf);
+size_t convo_info_volatile_size_legacy_closed(const config_object* conf);
+
+/// Functions for iterating through the entire conversation list.  Intended use is:
+///
+///     convo_info_volatile_1to1 c1;
+///     convo_info_volatile_open c2;
+///     convo_info_volatile_legacy_closed c3;
+///     convo_info_volatile_iterator *it = convo_info_volatile_iterator_new(my_convos);
+///     for (; !convo_info_volatile_iterator_done(it); convo_info_volatile_iterator_advance(it)) {
+///         if (convo_info_volatile_it_is_1to1(it, &c1)) {
+///             // use c1.whatever
+///         } else if (convo_info_volatile_it_is_open(it, &c2)) {
+///             // use c2.whatever
+///         } else if (convo_info_volatile_it_is_legacy_closed(it, &c3)) {
+///             // use c3.whatever
+///         }
+///     }
+///     convo_info_volatile_iterator_free(it);
+///
+/// It is permitted to modify records (e.g. with a call to one of the `convo_info_volatile_set_*`
+/// functions) and add records while iterating.
+///
+/// If you need to remove while iterating then usage is slightly different: you must advance the
+/// iteration by calling either convo_info_volatile_iterator_advance if not deleting, or
+/// convo_info_volatile_iterator_erase to erase and advance.  Usage looks like this:
+///
+///     convo_info_volatile_1to1 c1;
+///     convo_info_volatile_iterator *it = convo_info_volatile_iterator_new(my_convos);
+///     while (!convo_info_volatile_iterator_done(it)) {
+///         if (convo_it_is_1to1(it, &c1)) {
+///             bool should_delete = /* ... */;
+///             if (should_delete)
+///                 convo_info_volatile_iterator_erase(it);
+///             else
+///                 convo_info_volatile_iterator_advance(it);
```

Comment:
this advancement of the iterator should be done outside the `if(convo_it_is_1to1(...` block as well.

Reply:
Fixed in #15